internal/layer_api_5.go

```diff
@@ -0,0 +1,16 @@
+package internal
+
+// LayerAPI5 is used for backwards compatibility with serialization/deserialization of the layer toml
+type LayerAPI5 struct {
+	// Build indicates that a layer should be used for builds.
+	Build bool `toml:"build"`
+
+	// Cache indicates that a layer should be cached.
+	Cache bool `toml:"cache"`
+
+	// Launch indicates that a layer should be used for launch.
+	Launch bool `toml:"launch"`
```

Comment:
It might be easier to keep this as LayerTypes and embed LayerTypes in LayerAPI5.
This was what was done previously in api v0.5. This should make the conversions between v0.5 and v0.6 easier?

See https://github.com/buildpacks/libcnb/blob/ab7d1d48e082a754434ce52c941ece92194e4149/layer.go#L73

Reply:
I tried that initially, but it creates a circular package reference, internal -> libcnb -> internal -> ... 

I broke that by just including the three bools directly. It's not DRY, but this data structure will go away with the next API version, so it doesn't seem too awful. I also wanted to put this struct into the internal package to keep someone from trying to use it directly. I could move it next to the Layer struct in layer.go and use LayerTypes embedded. It seems unlikely that with a doc comment, someone would use it directly. 

Let me know if you have a preference, I'm OK with either one.

Reply:
Done